File: app/src/main/java/com/anokmik/databinding/component/MainDataBindingComponent.java
package com.anokmik.databinding.component;

import android.databinding.BindingAdapter;
import android.graphics.drawable.ColorDrawable;
import android.graphics.drawable.StateListDrawable;
import android.widget.Button;

public class MainDataBindingComponent {

    @BindingAdapter(value = {"defaultColor", "pressedColor"})
    public void setButtonStateListBackground(Button button, int defaultColor, int pressedColor) {
        button.setBackground(getButtonStateListDrawable(defaultColor, pressedColor));
    }

    private StateListDrawable getButtonStateListDrawable(int defaultColor, int pressedColor) {
        StateListDrawable drawable = new StateListDrawable();
        drawable.addState(new int[]{android.R.attr.state_pressed}, new ColorDrawable(pressedColor));
        drawable.addState(new int[]{}, new ColorDrawable(defaultColor));
        return drawable;
    }

}

File: app/src/main/kotlin/com/anokmik/databinding/component/MainDataBindingComponent.kt
package com.anokmik.databinding.component

import android.databinding.BindingAdapter
import android.graphics.drawable.ColorDrawable
import android.graphics.drawable.StateListDrawable
import android.widget.Button

class MainDataBindingComponent {

    @BindingAdapter(value = *arrayOf("defaultColor", "pressedColor"))
    fun setButtonStateListBackground(button: Button, defaultColor: Int, pressedColor: Int) {
        button.background = getButtonStateListDrawable(defaultColor, pressedColor);
    }

    private fun getButtonStateListDrawable(defaultColor: Int, pressedColor: Int): StateListDrawable {
        val drawable = StateListDrawable()
        drawable.addState(IntArray(1, { android.R.attr.state_pressed }), ColorDrawable(pressedColor));
        drawable.addState(IntArray(0), ColorDrawable(defaultColor));
        return drawable;
    }

}